### PR TITLE
feat(apple): add extended whitespace support to AppDelegate detection; add tests for code-tools

### PR DESCRIPTION
**Pull request by philprime**
- Adds tests for the code tools used to insert the setup snippet
- Adds more support for white-spacing in AppDelegate method detection

### CODE REVIEW

File: src/apple/code-tools.ts [hunk 1]
```diff
@@ -97,6 +97,7 @@ export function addCodeSnippetToProject(
   if (!match) {
     const swiftUIMatch = swiftUIRegex.exec(fileContent);
     if (!swiftUIMatch) {
+      // This branch is not reached, because we already checked for SwiftUI in isAppDelegateFile
```
Comment (philipphofmann) [inline src/apple/code-tools.ts:100]: `m`: Why do we need the check then?

File: test/apple/code-tools.test.ts [hunk 2]
```diff
@@ -1,20 +1,1038 @@
...
+
+@implementation AppDelegate
+
+- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
+    // Override point for customization after application launch.
+    return YES;
+}
+
+@end`;
+const validAppDelegateObjcWithSentry = `@import Sentry;
```
Comment (philipphofmann) [inline test/apple/code-tools.test.ts:75]: `l`
```suggestion
const validAppDelegateObjCWithSentry = `@import Sentry;
```

File: test/apple/code-tools.test.ts [hunk 3]
```diff
@@ -1,20 +1,1038 @@
...
+                // -- Act --
+                const result = exportForTesting.isAppDelegateFile(filePath);
+
+                // -- Assert --
+                expect(result).toBeTruthy();
+              });
+            });
+          }
+
+          describe('invalid cases', () => {
```
Comment (philipphofmann) [inline test/apple/code-tools.test.ts:246]: `m`: What about adding some cases for garbage content in the files, such as `adsfasdf` or what if there is a missing `}` in the app delegate, so the code doesn't compile.

**Review by philipphofmann — COMMENTED** [review]
Thanks for the extensive test suite. I added some suggestions to cover a few more edge cases.

File: src/apple/code-tools.ts (hunk 1 above)
Comment (philprime) [reply src/apple/code-tools.ts:100]: The compiler requires the null check, I can probably get rid of it, but only by refactoring the implementation. This PR focuses on adding initial test coverage

File: test/apple/code-tools.test.ts (hunk 3 above)
Comment (philprime) [reply test/apple/code-tools.test.ts:246]: Good feedback, it seems I made a copy-paste error here (so many tests 😅). I'll add a garbage example example but we actually don't care about compiling here, because we are not interpreting code, only character matching

Comment (philipphofmann) [reply test/apple/code-tools.test.ts:246]: I also don't care too much about compiling here, but I care about the wizard not crashing when somebody has garbage in their AppDelegate 😄

Comment (philprime) [reply test/apple/code-tools.test.ts:246]: I agree, but there is no difference in the test equivalence classes, so it would just result in multiple tests testing for the same thing.

I added something for completeness sake, but just wanted to point this out.

Comment (philipphofmann) [reply test/apple/code-tools.test.ts:246]: I disagree, the `asdasdf garbage` test is an extra equivalence class when you don't know the code. When you know the code it's not, though. I think it adds value. Thanks.

**Review by philipphofmann — APPROVED** [review]
LGTM
